core/src/main/scala/kafka/log/LogCleaner.scala

```diff
@@ -134,6 +134,13 @@ class LogCleaner(val config: CleanerConfig,
   }
 
   /**
+   * Truncate cleaner offset checkpoint for the given partition if its checkpointed offset larger than the given offset
+   */
+  def truncateCheckpoint(dataDir: File, topicAndPartition: TopicAndPartition, offset: Long) = {
```

Comment:
We should use explicit result types for top-level methods. In this case, it looks like the result type is `Unit`, so you can simply use procedure syntax (ie remove `=`).